pyquil/api/_qpu.py

```diff
@@ -168,10 +168,17 @@ def run(self, run_priority: Optional[int] = None):
         else:
             bitstrings = None
 
-        self._bitstrings = bitstrings
+        self._memory_results = dict()
```

Comment:
Why do you want a regular `dict` here, but a `defaultdict` in the parent class?

Reply:
Unless the intention was to have reading from undefined memory regions raise KeyError on the QPU?  I assume not since you changed it, but asking just in case I faked you out with my previous comment :)

Reply:
I don't think it was intentional: I think I wrote `dict` in both places, hit an error during testing, replaced one of the `dict`s with a `defaultdict`, forgot about the other.